src/Castle.Core/DynamicProxy/Generators/BaseClassProxyGenerator.cs

```diff
@@ -102,7 +102,9 @@ protected sealed override Type GenerateType(string name, INamingScope namingScop
 			// Crosses fingers and build type
 
 			var proxyType = emitter.BuildType();
+#if !NET9_0_OR_GREATER
 			InitializeStaticFields(proxyType);
+#endif
```

Comment:
This must be skipped here because the generated type cannot be activated while tied to a `PersitedAssemblyBuilder`.